source/app/controllers/urls_controller.rb

```diff
@@ -1,2 +1,39 @@
 class UrlsController < ApplicationController
+  def create
+    @url = Url.new(urls_params)
+
+    if @url.save
+      redirect_to urls_path
+    else
+      render 'new'
+    end
+  end
+
+  def index
+    @urls = Url.all
+  end
+
+  def new
+    @url = Url.new
+  end
+
+  def route_me
+    url = Url.find_by_short_url params[:short_url]
+
+    if url.nil?
```

Comment:
I prefer `if url` (drop `.nil?`) to phrase things positively, and put the "happy path" code in the top of the `if/else`.